### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by SukanyaSen530**
Pages -
- Login Page
- Sign Up page

Components -
- Input Field
- Alert
- Prevented & Protected Routes

Functionalities - 
- Input Validation
- Conditional routing

### CODE REVIEW

File: src/components/Alert/Alert.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    globalHandlers: { closeAlert },
+  } = useGlobalContext();
+
+  useEffect(() => {
+    if (show) {
+      setTimeout(() => closeAlert(), 3000);
+    }
+  }, [show]);
+
+  let alertTypeClass = "alert—info";
```
Comment (sharathnair9999) [inline src/components/Alert/Alert.jsx:21]: Liked the way you wrote this variable to manage the class of alert

**Review by sharathnair9999 — APPROVED** [review]
The code looks so clean and subtle. Good to go!

File: src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -60,6 +76,25 @@ const Navbar = () => {
...
           ))}
+
+          {!token ? (
+            <>
+              <p className="divider"></p>
+              {authNavigation.map((item) => (
+                <NavLink
+                  key={item.id}
+                  to={`${item.path}`}
+                  onClick={() => setShow(false)}
```
Comment (JowelTisso) [inline src/components/Navbar/Navbar.jsx]: Template literal is redundant over here

**Review by JowelTisso — APPROVED** [review]
Looks Good

**Review by biswanath-tewari — APPROVED** [review]

File: src/components/Navbar/Navbar.jsx (hunk 2 above)
Comment (SukanyaSen530) [reply src/components/Navbar/Navbar.jsx]: Fixed - [929526e](https://github.com/SukanyaSen530/DrawFlix/pull/2/commits/929526e5c941ee1136f78cf1716699dc2173096a)

**Comment by JowelTisso** [comment]
* Try to keep your PR small as it makes it easier to review

**Pull request merged**